lib/gstreamer/vppbase.py

```diff
@@ -50,6 +51,7 @@ def gen_input_opts(self):
     return opts
 
   def gen_output_opts(self):
+    self.osdecoded = filepath2os(self.decoded)
```

Comment:
This should only be done in the `vpp()` function each time after `self.decoded` is assigned.

Reply:
I know the current design can be confusing to know when and where certain variables should be accessed.  I'm hoping to clean this confusion up in some upcoming refactors soon.  In the meantime, we just have to be careful.